infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/UniqueWorkspacePVCStrategy.java

```diff
@@ -106,42 +125,47 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       ephemeralWorkspaceAdapter.provision(k8sEnv, identity);
       return;
     }
+
     LOG.debug("Provisioning PVC strategy for workspace '{}'", workspaceId);
+
+    fillInExistingPVCs(k8sEnv, workspaceId);
+
     // fetches all existing PVCs related to given workspace and groups them by volume name
     final Map<String, PersistentVolumeClaim> volumeName2PVC =
-        groupByVolumeName(
-            factory
-                .create(workspaceId)
-                .persistentVolumeClaims()
-                .getByLabel(CHE_WORKSPACE_ID_LABEL, workspaceId));
+        groupByVolumeName(k8sEnv.getPersistentVolumeClaims().values());
```

Comment:
I think `groupByVolumeName` needs more clear documentation on how it is used (i.e. that its purpose is to map user-defined PVCs on to previously created PVCs). It's clear *what* it does, but it took me a long time to understand *why* it does it.

Reply:
Added